lib/data-table/table.rb

```diff
@@ -339,7 +339,9 @@ def calculate_subtotals!
           @collection.each_pair_with_parents(@groupings.count) do |group_name, group_data, parents|
             path = parents + [group_name]
             result = calculate(group_data, subtotal[0], subtotal[1], path)
-            @subtotal_calculations[path][index] ||= {}
+            (0..index).each do |index|
```

Comment:
@aharpervc  What the reasoning for iterating over the range here? `(0..index)`

Reply:
Same logic as https://github.com/veracross/data-table/pull/6/files#diff-6d5c93afbede304616a4170b9f9dacadR443, which is that you need to fill in any prior array indexes with empty objects to avoid nil reference exceptions.

```ruby
x = []
x[1] = {}
puts x.inspect # [nil, {}]`
```

Reply:
were nil references the original problem that triggered this change?

Reply:
Yep, see https://support.veracross.com/a/tickets/17520 for the original test case, now followed by all of brian's examples, plus https://github.com/axiom-dev/planning/issues/1274